kernel/packages/unity-interface/BrowserInterface.ts

```diff
@@ -500,17 +501,21 @@ export class BrowserInterface {
     if (realm) {
       catalystRealmConnected().then(
         () => {
-          TeleportController.goTo(x, y, `Jumped to ${x},${y} in realm ${realmString}!`)
+          const successMessage = `Jumped to ${x},${y} in realm ${realmString}!`
+          notifyStatusThroughChat(successMessage)
+          unityInterface.ConnectionToRealmSuccess(data)
+          TeleportController.goTo(x, y, successMessage)
         },
         (e) => {
           const cause = e === 'realm-full' ? ' The requested realm is full.' : ''
           notifyStatusThroughChat('Could not join realm.' + cause)
-
+          unityInterface.ConnectionToRealmFailed(data)
           defaultLogger.error('Error joining realm', e)
         }
       )
     } else {
-      notifyStatusThroughChat(`Couldn't find realm ${realmString}`)
+      notifyStatusThroughChat(`Couldn't find realm ${realmString}.`)
```

Comment:
Note that in the case the retry system has to try multiple times, many `Couldn't find realm` chat messages may appear on the log. This may be undesired behaviour.